.github/scripts/manage-vm.py

```diff
@@ -144,7 +156,52 @@ def get_runner_token(github_repo_owner, github_repo, github_token):
         raise ValueError(f"Failed to get runner registration token: {result}")
 
 
-def create_vm(sdk, args):
+def wait_for_runner_registration(
+    client: Github,
+    vm_id: str,
+    github_repo_owner: str,
+    github_repo: str,
+    timeout_sec: int = 10,
+    retries: int = 60,
+) -> Optional[str]:
+    logger.info(
+        "Waiting for runner registration every %d seconds x %d times",
+        timeout_sec,
+        retries,
+    )
+
+    for i in range(retries):
+        runner_id = find_runner_by_name(client, github_repo_owner, github_repo, vm_id)
+        if runner_id is not None:
+            return runner_id
+        else:
+            time.sleep(timeout_sec)
+
+    return None
+
+
+def find_runner_by_name(
+    client: Github, github_repo_owner: str, github_repo: str, vm_id: str
+) -> Optional[str]:
+    runners = client.get_repo(
+        f"{github_repo_owner}/{github_repo}"
+    ).get_self_hosted_runners()
+
+    runner_id = None
+    for runner in runners:
+        if runner.name == vm_id:
+            runner_id = runner.id
+            break
+
+    if runner_id is None:
```

Comment:
Just log the vm_id and runner_id and return runner_id. The logging severity is the same. No need for additional checks/logs.